keps/sig-node/4680-add-resource-health-to-pod-status/README.md

```diff
@@ -117,60 +124,104 @@ The proposal is to keep this structure as-is to simplify parsing of well-known R
+	ResourceHealthStatusUnhealthy ResourceHealthStatus = "Unhealthy"
+	ResourceHealthStatusUnknown   ResourceHealthStatus = "Unknown"
+)
+
 // ResourceID is calculated based on the source of this resource health information.
 // For DevicePlugin:
-//   deviceplugin:DeviceID, where DeviceID is from the Device structure of DevicePlugin's ListAndWatchResponse type: https://github.com/kubernetes/kubernetes/blob/eda1c780543a27c078450e2f17d674471e00f494/staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1alpha/api.proto#L61-L73
+//
+//	DeviceID, where DeviceID is from the Device structure of DevicePlugin's ListAndWatchResponse type: https://github.com/kubernetes/kubernetes/blob/eda1c780543a27c078450e2f17d674471e00f494/staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1alpha/api.proto#L61-L73
+//
 // DevicePlugin ID is usually a constant for the lifetime of a Node and typically can be used to uniquely identify the device on the node.
 // For DRA:
-//   dra:<driver name>/<pool name>/<device name>: such a device can be looked up in the information published by that DRA driver to learn more about it. It is designed to be globally unique in a cluster.
+//
+//	<driver name>/<pool name>/<device name>: such a device can be looked up in the information published by that DRA driver to learn more about it. It is designed to be globally unique in a cluster.
 type ResourceID string
 
+// ResourceHealth represents the health of a resource. It has the latest device health information.
+// This is a part of KEP https://kep.k8s.io/4680 and historical health changes are planned to be added in future iterations of a KEP.
 type ResourceHealth struct {
-    // List of conditions with the transition times
-    Conditions []ResourceHealthCondition
+	// ResourceID is the unique identifier of the resource. See the ResourceID type for more information.
+	ResourceID ResourceID `json:"resourceID" protobuf:"bytes,1,opt,name=resourceID"`
+	// Health of the resource.
+	// can be one of:
+	//  - Healthy: operates as normal
+	//  - Unhealthy: reported unhealthy. We consider this a temporary health issue
+	//               since we do not have a mechanism today to distinguish
+	//               temporary and permanent issues.
+	//  - Unknown: The status cannot be determined.
+	//             For example, Device Plugin got unregistered and hasn't been re-registered since.
+	//
+	// In future we may want to introduce the PermanentlyUnhealthy Status.
+	Health ResourceHealthStatus `json:"health,omitempty" protobuf:"bytes,2,name=health"`
 }
+```
+
+In alpha2 in order to support pod level DRA resources, the following field will be added to the PodStatus:
+
+```go
+// PodStatus represents information about the status of a pod. Status may trail the actual
+// state of a system.
+type PodStatus struct {
+
+	...
+
+	// Status of resource claims.
+	// +featureGate=DynamicResourceAllocation
+	// +optional
+	ResourceClaimStatuses []PodResourceClaimStatus
```

Comment:
I am confused why we need this as well as `AllocatedResourceStatus` below? Also `PodResourceClaimStatus` type is not defined here?

I think container-level DRA claim status will land in the `ContainerStatus.AllocatedResourceStatus` entry with `name` of `claims`, and in `PodStatus` we actually only need something like`AllocatedResourceClaimHealth []ResourceHealth` since if we had `[]ResourceStatus` there would only ever be one entry for `claims`.

Or am I confused?

Reply:
PodResourceClaimStatus is listed as FYI. Mostly to show that we are not touching it for this KEP.

I don't want to assume that only claims will be available on Pod level in future. So keeping it open for potential expansion in future.